src/cement_ai_platform/data/processors/optimization_bayesian_hyperparameter_tuning.py

```
@@ -0,0 +1,267 @@
+colors = ['green' if imp > 0 else 'red' for imp in improvements]
+bars = ax2.bar(range(len(models)), improvements, color=colors, alpha=0.8)
+ax2.set_xlabel('Models')
+ax2.set_ylabel('R² Improvement')
+ax2.set_title('Performance Improvement from Tuning', fontweight='bold')
+ax2.set_xticks(range(len(models)))
+ax2.set_xticklabels(models, rotation=45, ha='right')
+ax2.axhline(y=0, color='black', linestyle='-', alpha=0.3)
+ax2.grid(True, alpha=0.3)
+
+# Add improvement values on bars
+for i, bar in enumerate(bars):
```

Ruff (0.13.1)

Loop control variable `i` not used within loop body

(B007)